Makefile

```diff
@@ -357,204 +449,55 @@ racetest: pilot-racetest mixer-racetest security-racetest broker-racetest galley
-docker.pilot: pilot/docker/pilot-discovery
-docker.proxy docker.proxy_debug: pilot/docker/pilot-agent ${PROXY_JSON_FILES}
-docker.proxy_init: pilot/docker/prepare_proxy.sh
-docker.sidecar_injector: pilot/docker/sidecar-injector
-
-PILOT_DOCKER:=docker.app docker.eurekamirror docker.pilot docker.proxy \
-              docker.proxy_debug docker.proxy_init docker.sidecar_injector
-$(PILOT_DOCKER): pilot/docker/Dockerfile$$(suffix $$@)
-	time (cd pilot/docker && docker build -t $(subst docker.,,$@) -f Dockerfile$(suffix $@) .)
-
-# mixer/example docker images
-
-SERVICEGRAPH_DOCKER:=docker.servicegraph docker.servicegraph_debug
-$(SERVICEGRAPH_DOCKER): mixer/example/servicegraph/docker/Dockerfile$$(if $$(findstring debug,$$@),.debug) \
-		mixer/example/servicegraph/docker/servicegraph mixer/example/servicegraph/docker/viz
-	time (cd mixer/example/servicegraph/docker && docker build -t servicegraph$(findstring _debug,$@) \
-		-f Dockerfile$(if $(findstring debug,$@),.debug) .)
-
-# mixer docker images
-
-MIXER_DOCKER:=docker.mixer docker.mixer_debug
-$(MIXER_DOCKER): mixer/docker/Dockerfile$$(if $$(findstring debug,$$@),.debug) \
-		mixer/docker/ca-certificates.tgz mixer/docker/mixs
-	time (cd mixer/docker && docker build -t mixer$(findstring _debug,$@) \
-		-f Dockerfile$(if $(findstring debug,$@),.debug) .)
-
-# security docker images
-
-docker.istio-ca: security/docker/istio_ca security/docker/ca-certificates.tgz
-docker.istio-ca-test: security/docker/istio_ca.crt security/docker/istio_ca.key
-docker.node-agent:      security/docker/node_agent
-docker.node-agent-test: security/docker/node_agent ${NODE_AGENT_TEST_FILES}
-
-SECURITY_DOCKER:=docker.istio-ca docker.istio-ca-test docker.node-agent docker.node-agent-test
-$(SECURITY_DOCKER): security/docker/Dockerfile$$(suffix $$@)
-	time (cd security/docker && docker build -t $(subst docker.,,$@) -f Dockerfile$(suffix $@) .)
-
-# grafana image
-
-docker.grafana: mixer/deploy/kube/conf/Dockerfile $(GRAFANA_FILES)
-	time (cd mixer/deploy/kube/conf && docker build -t grafana -f Dockerfile .)
-
-DOCKER_TARGETS:=$(PILOT_DOCKER) $(SERVICEGRAPH_DOCKER) $(MIXER_DOCKER) $(SECURITY_DOCKER) docker.grafana
-
-docker.all: $(DOCKER_TARGETS)
-
-# for each docker.XXX target create a tar.docker.XXX target that says how
-# to make a $(OUT)/docker/XXX.tar.gz from the docker XXX image
-# note that $(subst docker.,,$(TGT)) strips off the "docker." prefix, leaving just the XXX
-$(foreach TGT,$(DOCKER_TARGETS),$(eval tar.$(TGT): $(TGT); \
-   time (mkdir -p ${OUT}/docker && \
-         docker save -o ${OUT}/docker/$(subst docker.,,$(TGT)).tar $(subst docker.,,$(TGT)) && \
-         gzip ${OUT}/docker/$(subst docker.,,$(TGT)).tar)))
+# for now docker is limited to Linux compiles
+ifeq ($(GOOS),linux)
 
-# create a DOCKER_TAR_TARGETS that's each of DOCKER_TARGETS with a tar. prefix
-DOCKER_TAR_TARGETS:=
-$(foreach TGT,$(DOCKER_TARGETS),$(eval DOCKER_TAR_TARGETS+=tar.$(TGT)))
+include tools/istio-docker.mk
```

Comment:
can we put these in bin/ ? Tools has some code as well. bin is all scripts.

Reply:
When Costin requested I put the docker stuff in a separate mk he suggested this name/location.  These mk aren't quite executable either so should we start a "make/"?  Other possible candidates to move to this directory (if used) would be tests/istio.mk and tools/deb/istio.mk.